### ISSUE TITLE
[3.5.0] undefined symbol timelib_update_ts

### ISSUE THREAD

**Issue by remicollet** [google/protobuf#3929]
Tyring to load the 3.5.0 extension
`PHP Warning:  PHP Startup: Unable to load dynamic library '/dev/shm/BUILDROOT/php-pecl-protobuf-3.5.0-1.fc25.remi.7.0.x86_64/usr/lib64/php/modules/protobuf.so' - /dev/shm/BUILDROOT/php-pecl-protobuf-3.5.0-1.fc25.remi.7.0.x86_64/usr/lib64/php/modules/protobuf.so: undefined symbol: timelib_update_ts in Unknown on line 0
`

This is NOT a public api, so cannot be used.

**Comment by remicollet** [comment]
@derick, mostly FYI

BTW, does this make sense to install timelib.h header if functions are not public ?
Perhaps some may be made public, other move to private header ?

**Comment by remicollet** [comment]
@derickr mostly FYI

BTW, does this make sense to install timelib.h header if functions are not public ?
Perhaps some may be made public, other move to private header ?

**Comment by remicollet** [comment]
Notice, they are indeed public from timelib PoV, but private from PHP PoV

**Comment by remicollet** [comment]
3.5.0.1 still broken

**Comment by TeBoring** [comment]
@remicollet thanks for help. What's Pov by the way?

**Comment by TeBoring** [comment]
I have a question. If user has installed the date extension, is timelib_update_ts guaranteed to be available?
If yes, I will just add a check in protobuf (throw error if user want to call Timestamp method without installing data extension). If not, I will implement using other API. @remicollet

**Comment by TeBoring** [comment]
@remicollet Do u mind take a look at #4034?

**Comment by remicollet** [comment]
> I have a question. If user has installed the date extension, is timelib_update_ts guaranteed to be available?

1/ the date extension is always there (cannot be shared or disabled)
2/ timelib_update_ts  is never there (private)

> Do u mind take a look at #4034?

I will try the patch later this week (busy with 7.2.1RC1)

btw, could make sense to add a dependency on date extension (using ZEND_MOD_REQUIRED)

**Comment by remicollet** [comment]
> Do u mind take a look at #4034?

Tested 3.5.0.1 + pr #4034 
Build + load test OK with 5.5, 5.6, 7.0, 7.1 and 7.2

Thanks

**Comment by remicollet** [comment]
@do you plan a quick bugfix release (3.5.0.2) or should I use this patch to provide 3.5.0.1 in my [repository](https://blog.remirepo.net/pages/PECL-extensions-RPM-status) (which is stalled at 3.4 for now)  ?

**Comment by TeBoring** [comment]
We will release 3.5.1 to include this change by end of week.

**Comment by remicollet** [comment]
> We will release 3.5.1 to include this change by end of week.

Thanks, will wait for 3.5.1 then.

**Comment by k4k** [comment]
Should building from `master/HEAD` resolve this issue? I ask because I've built master (which presumably has the fix to be included in 3.5.1) and am still receiving the following message:

```
PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/lib64/php/modules/protobuf.so' - /usr/lib64/php/modules/protobuf.so: undefined symbol: timelib_update_ts in Unknown on line 0
```

**Comment by remicollet** [comment]
I confirm, 3.5.1 is ok
